### PR TITLE
fire the OpenedChangeEvent for the opened-changed dom event

### LINKED ISSUE #5386: Date Picker addOpenedChangeListener not working
### Description of the bug

When using a Date-Picker and adding a OpenedChangeListener the Event will not be handled.

### Expected behavior

The added Listener should run the method defined in the lambda.

### Minimal reproducible example

https://github.com/DarioViva42/addOpenedChangeListener
Run the application using mvn.
Checkout branch withVaadin23 to experience the same project, but working.

### Versions

- Vaadin / Flow version: 24
- Java version: 17
- OS version: Windows 10

### PR DESCRIPTION
**Pull request by sujoykd**
## Description

Fixes #5386

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.
- [ ] I have not completed some of the steps above and my pull request can be closed immediately.

### CODE REVIEW

**Review by web-padawan — CHANGES_REQUESTED** [review]
Let's also check other components that might be affected, see https://github.com/vaadin/flow-components/issues/5386#issuecomment-1702694823

**Comment by sujoykd** [comment]
> Let's also check other components that might be affected, see [#5386 (comment)](https://github.com/vaadin/flow-components/issues/5386#issuecomment-1702694823)

I am checking the other components that might be affected.
However, the fix for #4909 was released on 24.0.5, but this current issue is also present in 24.0.4

File: vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java [hunk 1]
```diff
@@ -748,9 +748,13 @@ protected void close() {
 
     /**
      * Gets the states of the drop-down for the datepicker
+     * <p>
+     * This property is synchronized automatically from client side when an
+     * {@code opened-changed} event happens.
      *
      * @return {@code true} if the drop-down is opened, {@code false} otherwise
      */
+    @Synchronize(property = "opened", value = { "opened-changed" })
```
Comment (web-padawan) [inline vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:757]:
```suggestion
    @Synchronize(property = "opened", value = "opened-changed")
```

**Review by web-padawan — APPROVED** [review]

**Review by ugur-vaadin — APPROVED** [review]
I checked the other components as well. The `@Synchronize` annotation appears to be missing only from this one. It seems that it was mistakenly removed while removing the deprecated generated classes. I wonder if we should introduce (out of the scope of this PR&issue) similar tests for all the component listeners that depend on DOM events.

**Pull request merged**